### ISSUE TITLE
conda-forge build fails because of too long file name

### ISSUE THREAD

**Issue by ltalirz** [DropD/reentry#28]
See https://circleci.com/gh/conda-forge/reentry-feedstock/14?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

The issue occurs here:
https://github.com/DropD/reentry/blob/f1cd3bfc0a9af5d2c31fa7a79b9a4208c3ddc019/reentry/config.py#L73

Error message:
```
py.error.ENAMETOOLONG: [File name too long]: open('/home/conda/.config/reentry/data/home.conda.feedstock_root.build_artifacts.reentry_1527279672474._test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_.bin.python', 'w')
```
Once could argue this is more a problem of conda-build than of reentry, still it would be nice to fix it.

**Comment by DropD** [comment]
The solution for this might be to use an md5 hash instead of a mangled path for the name.

**Comment by ltalirz** [comment]
jep...

**Comment by ltalirz** [comment]
from version `1.2.0` onwards, `reentry` cannot even be *imported* in the conda test environment anymore...

**Comment by ltalirz** [comment]
Fixed in #37
